pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1386,13 +1386,23 @@ func doRestorePlan(
 		}
 	}
 
-	// Validate that the backup is a full cluster backup if a full cluster restore was requested.
-	if restoreStmt.DescriptorCoverage == tree.AllDescriptors && mainBackupManifests[0].DescriptorCoverage == tree.RequestedDescriptors {
-		return errors.Errorf("full cluster RESTORE can only be used on full cluster BACKUP files")
-	}
+	isClusterRestore := restoreStmt.DescriptorCoverage == tree.AllDescriptors
```

Comment:
nit: seems like we mostly use `if restoreStmt.DescriptorCoverage == tree.AllDescriptors` to branch off for cluster restore specific logic.